text/0017-shared-texture.md

```diff
@@ -0,0 +1,491 @@
+- Status: **Active**
+
+# Import Shared Texture
+
+## Summary
+
+This feature introduces a way to import a native shared texture handle into Electron, specifically as a [`VideoFrame`](https://developer.mozilla.org/en-US/docs/Web/API/VideoFrame). `VideoFrame` natively supports several web rendering systems, including `WebGPU` and `WebGL`. This enables developers to integrate arbitrary native-rendered content with their web applications.
+
+https://github.com/user-attachments/assets/b09bde3a-d2c5-46c3-b3ef-52dda994eb28
+
+## Motivation
+
+While `Offscreen Rendering` allows developers to export a shared texture for embedding the browser's output into their own rendering programs efficiently, there are cases where developers may want the reverse: to import images into the browser. For example, broadcasting apps may use this feature to embed various input sources with modern frontend technologies, providing a contemporary user experience. By using shared textures, we can maximize performance and provide a standardized way to bridge images between Electron and native rendering programs.
+
+## API Design
+
+### Module
+
+A new `sharedTexture` module will be added, available in both the sandboxed renderer process and the main process. This module provides the primary interface for importing external shared texture handles. The module handles lifetime management automatically, while still provide subtle API for advanced users. 
+
+### Methods
+
+#### `sharedTexture.importSharedTexture(options)` _Experimental_
+
+> This method is only available in the main process.
+
+* `options` Object - Options for importing shared texture.
+  * `textureInfo` [SharedTextureImportTextureInfo](structures/shared-texture-import-texture-info.md) - The information of shared texture to import.
+  * `allReferenceReleased` Function (optional) - Called when all references in all processes are released, you should keep native texture valid until this callback is called.
+
+Imports the shared texture from the given options. A copy of the native texture will be made, so that you can release the native texture after the import is complete.
+
+Returns `SharedTextureImported` - The imported shared texture.
+
+#### `sharedTexture.sendToRenderer(webContents, imported, ...args)` _Experimental_
+
+> This method is only available in the main process.
+
+* `webContents` [WebContents](https://github.com/electron/electron/blob/main/docs/api/web-contents.md) - The web contents to transfer the shared texture to.
+* `importedSharedTexture` [SharedTextureImported](#sharedtextureimported-object) - The imported shared texture.
+* `...args` any[] - Additional arguments to pass to the renderer process.
+
+Send the imported shared texture to a renderer process. You must register receiver at renderer process before calling this method. This method has a timeout.
+
+Returns `Promise<void>` - Resolves when the transfer is complete.
+
+#### `sharedTexture.receiveFromMain(receiver)` _Experimental_
```

Comment:
This name is a little confusing because if I understand correctly, you wouldn't call it each time you receive a texture, but would call it once to set up a listener.

What if we call this something like `sharedTexture.setTextureReceiver`?